engine/pong.h

```diff
@@ -36,19 +36,38 @@ class Pong
     static void Reset();
     static void Cleanup();
 
-    static void LoadSceneNext(Scene scene);
+    static void LoadSceneNext(SceneType scene);
 
-    template<typename T>
-    static T* FindGameObject()
+    // Returns the first component of type T found in the scene
+    template<typename T, typename = std::enable_if_t<std::is_base_of_v<BaseComponent, T> && !std::is_base_of_v<Behavior, T>>>
+    static T* FindComponentOfType()
     {
-        for (auto& gameObject : GetInstance().mGameObjects)
+        const std::vector<std::unique_ptr<T>>& components = T::GetComponents();
+
+        if (components.empty())
         {
-            auto castedGameObject = dynamic_cast<T*>(gameObject.get());
```

Comment:
Moving to the ECS and using CRTP also solved these dynamic casting issues. Very cool! 🎉